server/api/build.go

```diff
@@ -34,6 +38,48 @@ import (
 	"github.com/woodpecker-ci/woodpecker/server/store"
 )
 
+func CreateBuild(c *gin.Context) {
+	_store := store.FromContext(c)
+	repo := session.Repo(c)
+
+	var p woodpecker.BuildOptions
+
+	err := json.NewDecoder(c.Request.Body).Decode(&p)
+	if err != nil {
+		_ = c.AbortWithError(http.StatusBadRequest, err)
+		return
+	}
+
+	user := session.User(c)
+
+	lastCommit, _ := server.Config.Services.Remote.BranchHead(c, user, repo, p.Branch)
+
+	tmpBuild := &model.Build{
```

Comment:
can we move create Build out of  CreateBuildOptions into it's own encapsulated func so we can unit test it?